### PR TITLE
GEODE-4830: list jndi binding command should list what's in the cluster configuration as well.

### PR DESCRIPTION
**Pull request by jinmeiliao**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jinmeiliao** [comment]
precheckin green

File: geode-core/src/main/java/org/apache/geode/internal/cache/configuration/CacheConfig.java [hunk 1]
```diff
@@ -0,0 +1,3570 @@
...
+ *
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "",
+    propOrder = {"cacheTransactionManager", "dynamicRegionFactory", "gatewayHub", "gatewaySender",
+        "gatewayReceiver", "gatewayConflictResolver", "asyncEventQueue", "cacheServer", "pool",
+        "diskStore", "pdx", "regionAttributes", "jndiBindings", "region", "functionService",
+        "resourceManager", "serializationRegistration", "backup", "initializer", "cacheElements"})
+@XmlRootElement(name = "cache", namespace = "http://geode.apache.org/schema/cache")
+public class CacheConfig {
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/configuration/CacheConfig.java]: Can we call this `CacheConfigType` for consistency?

File: geode-core/src/main/java/org/apache/geode/internal/cache/configuration/ObjectFactory.java [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+ * for XML content. The Java representation of XML
+ * content can consist of schema derived interfaces
+ * and classes representing the binding of schema
+ * type definitions, element declarations and model
+ * groups. Factory methods for each of these are
+ * provided in this class.
+ *
+ */
+@XmlRegistry
+public class ObjectFactory {
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/internal/cache/configuration/ObjectFactory.java]: We probably don't need this class.

File: geode-core/src/main/java/org/apache/geode/internal/cache/configuration/CacheConfig.java (hunk 1 above)
Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/configuration/CacheConfig.java]: Added a subtask under our make CC public story to move that entire package to a public package and consider renaming of its classes.

File: geode-core/src/main/java/org/apache/geode/internal/cache/configuration/ObjectFactory.java (hunk 2 above)
Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/configuration/ObjectFactory.java]: done

**Pull request closed without merging**